src/usageReport.ts

```diff
@@ -0,0 +1,55 @@
+import { v4 as uuidv4 } from 'uuid'
+import { app } from 'electron'
+import { getPlatform, getArch } from './utils/electron'
+import { writeFile, readFile } from 'fs/promises'
+import path from 'path'
+import { existsSync } from 'fs'
+import { UsageReportSettings } from './types/settings'
+
+const URL = 'https://stats.grafana.org/k6-studio-usage-report'
+const filename = '.installation_id'
+const filePath = path.join(app.getPath('userData'), filename)
+
+export const getOrSetInstallationId = async () => {
+  if (!existsSync(filePath)) {
+    const id = uuidv4()
+    await writeFile(filePath, id)
+    return id
+  }
+
+  return await readFile(filePath, { encoding: 'utf-8' })
+}
+
+export const sendReport = async (usageReportSettings: UsageReportSettings) => {
+  if (!usageReportSettings.enabled) {
+    return
+  }
+
+  if (process.env.NODE_ENV === 'development' || existsSync(filePath)) {
+    // if we sent the report for this installation, do nothing
+    return
+  }
+
+  // Update the Usage Report section in README when changing what data is being sent
+  // https://github.com/grafana/k6-studio/blob/main/README.md#usage-report
+  const id = await getOrSetInstallationId()
+  const data = {
+    usageStatsId: id,
+    timestamp: Date(),
+    os: getPlatform(),
+    arch: getArch(),
+    version: app.getVersion(),
+  }
+
+  try {
+    await fetch(URL, {
+      method: 'post',
+      body: JSON.stringify(data),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+  } catch (e) {
+    console.log('Error sending report', e)
```

Comment:
Since we don't send the report when running the app in dev env, this line isn't really needed. I think we should either log it in the app logs, or fail silently instead